code/modules/admin/verbs/greytide_worldwide.dm

```diff
@@ -0,0 +1,152 @@
+			mobs_to_convert += M
+
+	for(var/mob/M in mobs_to_convert)
+		event.convert_mob(M)
+
+	event.event_setup_end(mobs_to_convert)
+
+	qdel(event)
+
+
+/datum/greytide_worldwide
+	var/name = "" //What to show in the list of choices. Does not appear in the list if left blank.
+	var/list/eligible_mobs //List of mobs to CHECK for eligibility. Not the final list of mobs to convert. Assigned in New() because BYOND a shit.
+
+//Only used for assigning eligible_mobs at the moment. It doesn't work in the type definition.
+/datum/greytide_worldwide/New()
+	eligible_mobs = player_list
+
+//Called at the beginning of the event. Cancels the event if it returns false, so use this to start anything you want to start immediately and/or check if the event can't happen.
+/datum/greytide_worldwide/proc/event_setup_start()
+	return 1
+
+//Used to retrieve the list of mobs to convert. Called on each mob in eligible_mobs. Any mob this returns true for is added to the final conversion list.
+/datum/greytide_worldwide/proc/check_eligibility(var/mob/M)
+	if(M.stat == DEAD)
+		return 0
+	if(ishuman(M) || issilicon(M))
+		return 1
+
+//Does the actual converting of mobs. Called on each mob in the final conversion list. Returns the converted mob, in case conversion requires spawning a new one.
+//The base version of this proc turns just turns silicons into humans, so any events that require that need only call the parent and convert the mob returned.
+/datum/greytide_worldwide/proc/convert_mob(var/mob/M)
+	if(issilicon(M))
+		var/mob/living/silicon/S = M
+		var/mob/living/carbon/human/new_human = new /mob/living/carbon/human(S.loc, delay_ready_dna=1)
+		new_human.setGender(pick(MALE, FEMALE)) //The new human's gender will be random
+		var/datum/preferences/A = new()	//Randomize appearance for the human
+		A.randomize_appearance_for(new_human)
+		new_human.generate_name()
+		new_human.languages |= S.languages
+		if(S.default_language)
+			new_human.default_language = S.default_language
+		if(S.mind)
+			S.mind.transfer_to(new_human)
+		else
+			new_human.key = S.key
+		qdel(S)
+		M = new_human
+	return M
+
+//Called at the end of the event setup. Provided with the list of converted mobs in case something has to be done with them after everything else is done.
+/datum/greytide_worldwide/proc/event_setup_end(var/list/converted_mobs)
+
+
+//The classic.
+/datum/greytide_worldwide/greytide
+	name = "Greytide Worldwide"
+
+/datum/greytide_worldwide/greytide/check_eligibility(var/mob/M)
+	if(!..())
+		return 0
+	if(is_special_character(M))
+		return 0
+	return 1
+
+/datum/greytide_worldwide/greytide/convert_mob(var/mob/M)
+	var/mob/living/carbon/human/H = ..(M)
+
+	ticker.mode.traitors += H.mind
+	H.mind.special_role = GREYTIDE // NEEDED FOR CHEAT CHECKS!
+
+	H.mutations.Add(M_HULK) //all greytiders are permahulks
+	H.set_species("Human", force_organs=TRUE) // No Dionae
+	H.a_intent = I_HURT
+
+	H.update_mutations()
+	H.update_body()
+
+	/* This never worked.
```

Comment:
so you're even copypasting pointless comments?

Reply:
yes, i'll clean all of them up at some point